integration/soroban/storage_types.spec.js

```diff
@@ -0,0 +1,87 @@
+import * as StellarSdk from '@stellar/stellar-sdk';
+import { readFileSync } from 'fs';
+import { expect } from 'chai';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { call_contract_function } from './test_helpers.js';  // Helper to interact with the contract
+
+const __filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(__filename);
+
+describe('StorageTypes', () => {
+  let keypair;
+  const server = new StellarSdk.SorobanRpc.Server(
+    "https://soroban-testnet.stellar.org:443",
+  );
+
+  let contractAddr;
+  let contract;
+  before(async () => {
+    console.log('Setting up storage_types contract tests...');
+
+    // Read secret from file
+    const secret = readFileSync('alice.txt', 'utf8').trim();
+    keypair = StellarSdk.Keypair.fromSecret(secret);
+
+    let contractIdFile = path.join(dirname, '.soroban', 'contract-ids', 'storage_types.txt');
+    // Read contract address from file
+    contractAddr = readFileSync(contractIdFile, 'utf8').trim().toString();
+
+    // Load contract
+    contract = new StellarSdk.Contract(contractAddr);
+
+    // Initialize the contract if necessary (adapt according to your contract logic)
+    await call_contract_function("init", server, keypair, contract);
+  });
+
+  it('check initial values', async () => {
+    // Check initial values of all storage variables
+    let sesa = await call_contract_function("sesa", server, keypair, contract);
```

Comment:
To expand testing coverage, consider adding a test case for scenarios where a variable **lacks** a specified storage type and defaults to `persistent`. This would ensure that the defaulting mechanism works as intended in Soroban contracts.


Also, what I said previously about `sesa` not being a descriptive variable.